bentoml/cli/bento.py

```diff
@@ -260,3 +270,100 @@ def retrieve(bento, target_dir):
+        help='Containerize given Bento into a Docker image',
+        short_help="Containerize given Bento into a Docker image",
+    )
+    @click.argument("bento", type=click.STRING)
+    @click.option('--push', is_flag=True)
+    @click.option(
+        '--docker-repository',
+        help="Prepends specified Docker repository to image name.",
+    )
+    def containerize(bento, push, docker_repository):
+        """Containerize specified BentoService.
+
+        BENTO is the target BentoService to be containerized, referenced by its name
+        and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+        `bentoml containerize` command also supports the use of the `latest` tag
+        and will automatically use the last built version of your Bento.
+
+        // add stuff about --push
+        // add stuff about --docker-repository
+        """
+        if ':' not in bento:
+            _echo(f'BentoService {bento} invalid - specify name:version')
+            return
+        name, version = bento.split(':')
+
+        yatai_client = YataiClient()
+
+        track_cli('bento-containerize')
+        get_bento_result = yatai_client.repository.get(name, version)
+        if get_bento_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                get_bento_result.status
+            )
+            _echo(
+                f'BentoService {name}:{version} not found - '
+                f'{error_code}:{error_message}',
+                CLI_COLOR_ERROR,
+            )
+            return
+
+        if get_bento_result.bento.uri.s3_presigned_url:
+            bento_service_bundle_path = get_bento_result.bento.uri.s3_presigned_url
+        else:
+            bento_service_bundle_path = get_bento_result.bento.uri.uri
+
+        _echo(f"Found Bento: {bento_service_bundle_path}")
+
+        if docker_repository is not None:
+            name = f'{docker_repository}/{name}'
+
+        name, version = _make_bento_name_docker_compatible(name, version)
+        tag = f"{name}:{version}"
+        if tag != bento:
+            _echo(
+                f'Bento name or tag was changed to be Docker compatible. \n'
+                f'"{bento}" -> "{tag}"',
+                CLI_COLOR_WARNING,
+            )
+
+        try:
+            docker_api = docker.APIClient()
+            ensure_docker_available_or_raise()
+            with Spinner(f"Building Docker image: {name} "):
+                _echo_docker_api_result(
+                    docker_api.build(
+                        path=bento_service_bundle_path, tag=tag, decode=True,
+                    )
+                )
+        except (FileNotFoundError, subprocess.CalledProcessError) as error:
+            _echo(
+                f'Could not build Docker image: {error}', CLI_COLOR_ERROR,
+            )
+            return
+
+        _echo(
+            f'Built {tag}', CLI_COLOR_SUCCESS,
```

Comment:
Let's provide better description. Maybe `Finished built {tag} for BentoService {bento}`